### PR TITLE
14 create squid indexer for pendulum network

### LINKED ISSUE #14: Create squid (indexer) for Pendulum network
So far we only have squids for Amplitude and Foucoco. Since we want to launch Nabla on Pendulum soon, we will need to have an indexer for Pendulum as well. 

Thus, we should create a pendulum-squid that has the same functionalities as the others
- Add pendulum to the [config](https://github.com/pendulum-chain/pendulum-squids/blob/main/src/config.ts) file
- Create a `squid-pendulum.yaml` file, similar to the others
- Create a `typegen-pendulum.json` file, similar to the others
- Add `process:pendulum` and `typegen:pendulum` to the [commands](https://github.com/pendulum-chain/pendulum-squids/blob/main/commands.json)
- Generate and use the pendulum-specific types
- Deploy it to aquarium

### PR DESCRIPTION
**Pull request by adelarja**
Issue #14

This PR has the basic configuration for the pendulum squid.

In the future, we could need to handle the farming pallet events for the pendulum chain too. A draft event handling was added to the farming utils [file](https://github.com/pendulum-chain/pendulum-squids/blob/b66c0f4ee6c02ee5ec67cdb9ff36288a10bcd5c6/src/utils/farming.ts#L29C7-L35C7) (for example [here](https://github.com/pendulum-chain/pendulum-squids/blob/b66c0f4ee6c02ee5ec67cdb9ff36288a10bcd5c6/src/utils/farming.ts#L29C7-L35C7)). Basically, the foucoco event types were used. That would need to be changed in the future, when the farming pallet is implemented on pendulum.

### CODE REVIEW

File: src/mappings/prices.ts [hunk 1]
```diff
@@ -1,16 +1,22 @@
 import { EventHandlerContext } from '../types'
-import { foucocoEvents, amplitudeEvents } from '../types/events'
+import { foucocoEvents, amplitudeEvents, pendulumEvents } from '../types/events'
 import { network } from '../config'
 import { getOrCreateOraclePrice } from '../entities/oraclePrice'
 
 export async function handleUpdatedPrices(ctx: EventHandlerContext) {
     let event
-    if (network !== 'foucoco') {
+    if (network === 'foucoco') {
```
Comment (adelarja) [inline src/mappings/prices.ts:8]: @ebma, I've changed this line. Is this okay, or were we intentionally using the `network !== 'foucoco'` comparison?

**Review by ebma — APPROVED** [review]
Nice looking good, thanks a lot @adelarja! 👍 

Can we have some 'FIXME' comment in the `farming.ts` file that points out that once the farming is liven on Pendulum, we need to regenerate the types and replace the `foucocoEvents.xxx` assignments for the pendulum `if`-statements?

File: src/mappings/prices.ts (hunk 1 above)
Comment (ebma) [reply src/mappings/prices.ts:8]: Nice catch 👍 this must have been a mistake and changing it to `===` should be correct.

**Comment by adelarja** [comment]
> Nice looking good, thanks a lot @adelarja! 👍
> 
> Can we have some 'FIXME' comment in the `farming.ts` file that points out that once the farming is liven on Pendulum, we need to regenerate the types and replace the `foucocoEvents.xxx` assignments for the pendulum `if`-statements?

@ebma thanks for the comment! I've added some `FIXME` comments.

**Comment by ebma** [comment]
Thanks @adelarja 👍 can we also add the instructions for the pendulum squid to the README? Sorry, I forgot to point this out last time 😅

**Comment by adelarja** [comment]
> Thanks @adelarja 👍 can we also add the instructions for the pendulum squid to the README? Sorry, I forgot to point this out last time 😅

@ebma thanks for pointing this out! I've added the instructions to the readme :smile:

**Review by ebma — APPROVED** [review]
Nice, thanks @adelarja! 
Ready for merge 👍

**Pull request merged**